pkg/validations/upgradevalidations/immutableFields.go

```diff
@@ -88,26 +88,26 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 		return fmt.Errorf("spec.externalEtcdConfiguration is immutable")
 	}
 
-	if !v1alpha1.RefSliceEqual(nSpec.IdentityProviderRefs, oSpec.IdentityProviderRefs) {
-		return fmt.Errorf("spec.identityProviderRefs is immutable")
+	oldAWSIamConfig := &v1alpha1.Ref{}
+
+	for _, oIdentityProvider := range oSpec.IdentityProviderRefs {
+		switch oIdentityProvider.Kind {
+		case v1alpha1.AWSIamConfigKind:
+			{
+				oldAWSIamConfig = &oIdentityProvider
+			}
+		}
 	}
-	if len(nSpec.IdentityProviderRefs) > 0 {
-		for _, nIdentityProvider := range nSpec.IdentityProviderRefs {
-			switch nIdentityProvider.Kind {
-			case v1alpha1.OIDCConfigKind:
-				prevOIDC, err := k.GetEksaOIDCConfig(ctx, nIdentityProvider.Name, cluster.KubeconfigFile, spec.Namespace)
-				if err != nil {
-					return err
-				}
-				if !prevOIDC.Spec.Equal(&spec.OIDCConfig.Spec) {
-					return fmt.Errorf("oidc identity provider is immutable")
-				}
-			case v1alpha1.AWSIamConfigKind:
+	for _, nIdentityProvider := range nSpec.IdentityProviderRefs {
```

Comment:
do we need two loops?

Reply:
the list on new identity providers vs old can change based on if OIDC settings were updated or not, so 2 loops are required.